uncertainties/core.py

```diff
@@ -332,7 +334,7 @@ def __setstate__(self, state):
         (self.linear_combo,) = state
 
 
-class AffineScalarFunc(object):
+class AffineScalarFunc(UFloatNumpy):
```

Comment:
This makes `AffineScalarFunc` and thus all of `uncertainties` dependent on numpy which I don't think has been discussed recently? The only previous discussion I can find is #47 where the conclusion seemed to be that improving numpy support should go into `uarray` rather than `AffineScalarFunc`/`UFloat` which makes sense to me.

Reply:
I agree with @wshanks. `AffineScalarFunc` should not require numpy.  
It is a scalar, it does not need broadcasted ufuncs.

Reply:
this is how to get np.sin(x) to work, removing the need for unumpy.sin
I meant to make UFloatNumpy an empty class when numpy is not installed. Got a bit trigger happy when finally getting the tests to pass! (The ufuncs get used when a `AffineScalarFunc`/`UFloat` gets multiplied, added etc by a np.array containing anything)

Reply:
we also wouldn't need a umath.sin, although that may be worth keeping so users don't need to have numpy installed

Reply:
@andrewgsavage I have not looked in detail at this.  I think I may not really understand your responses.  

Are you saying that the thing called `UFloatNumpy` will not depend on numpy?   That is going to be super-confusing.   Should this class use `numpy` if available and `math` otherwise?  Maybe call that `UMath`?

Again agreeing with @wshanks please you give an overview of the design goals and concepts, or a link to an earlier discussion.